packages/provider/src/multi-wallet.ts

```diff
@@ -0,0 +1,271 @@
+    return tx
+  }
+
+  async isDeployed(target?: Wallet | NetworkConfig): Promise<boolean> {
+    const wallet = (() => {
+      if (!target) return this.authWallet()
+      if ((<Wallet>target).address) {
+        return target as Wallet
+      }
+      return this.networkWallet(target as NetworkConfig)
+    })()
+
+    const walletCode = await wallet.provider.getCode(this.address)
+    return walletCode && walletCode !== "0x"
+  }
+
+  // TODO: Split this to it's own class "configProvider" or something
+  // this process can be done in different ways (caching, api, utils, etc)
+  async currentConfig(target?: Wallet | NetworkConfig): Promise<ArcadeumWalletConfig | undefined> {
+    const address = this.address
+    const wallet = (() => {
+      if (!target) return this.authWallet()
+      if ((<Wallet>target).address) {
+        return target as Wallet
+      }
+      return this.networkWallet(target as NetworkConfig)
+    })()
+  
+    const walletContract = new Contract(address, mainModuleUpgradableAbi, wallet.provider)
+
+    const authWallet = this.authWallet()
+    const authContract = new Contract(authWallet.context.requireUtils, requireUtilsAbi, authWallet.provider);
+
+    const currentImageHash = walletContract.functions.imageHash.call([])
+    currentImageHash.catch(() => {}) // Ignore no imageHash defined
+    const currentImplementation = ethers.utils.defaultAbiCoder.decode(
+      ['address'], await wallet.provider.getStorageAt(address, address)
+    )[0]
+
+    let event: any
+    if (currentImplementation === wallet.context.mainModuleUpgradable) {
+      // The wallet has been updated
+      // lookup configuration using imageHash
+      const filter = authContract.filters.RequiredConfig(null, await currentImageHash)
+      const logs = await authWallet.provider.getLogs({ fromBlock: 0, toBlock: 'latest', ...filter})
+      if (logs.length === 0) return undefined
+      const lastLog = logs[logs.length - 1]
+      event = authContract.interface.events.RequiredConfig.decode(lastLog.data, lastLog.topics)
+    } else {
+      // The wallet it's using the counter-factual configuration
+      const filter = authContract.filters.RequiredConfig(address)
+      const logs = await authWallet.provider.getLogs({ fromBlock: 0, toBlock: 'latest', ...filter})
+      if (logs.length === 0) return undefined
+      const lastLog = logs[0] // TODO: Search for real counter-factual config
+      event = authContract.interface.events.RequiredConfig.decode(lastLog.data, lastLog.topics)
+    }
+
+    const signers = ethers.utils.defaultAbiCoder.decode(
+      [`tuple(
+        uint256 weight,
+        address signer
+      )[]`], event._signers
+    )[0]
+
+    const config = {
+      address: address,
+      threshold: event._threshold,
+      signers: signers.map((s: any) => ({
+        address: s.signer,
+        weight: s.weight
+      }))
+    }
+
+    return config
+  }
+
+  private networkWallet(network: NetworkConfig | BigNumberish): Wallet {
```

Comment:
what is a "network wallet" ?

Reply:
This is a helper method used to retrieve the wallet of a given network, (an internal wallet it's created for each one of the networks).